btcwallet/btcwallet.go

```diff
@@ -0,0 +1,52 @@
+package btcwallet
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+// Bitcoin representation via the float64 data type.
+type Bitcoin float64
+
+// BtcWallet implements a simple bitcoins wallet for deposit, withdraw and get current balance.
+type BtcWallet struct {
+	sync.Mutex         // Embedded sync.Mutex that safe wallet for concurrent deposit and withdrawal operations.
+	balance    Bitcoin // balance stores the current value of the bitcoin(float64) balance in the wallet.
+}
+
+// NewBtcWallet returns a new BtcWallet which stores a predefined balance value.
+func NewBtcWallet(balance Bitcoin) *BtcWallet {
+	return &BtcWallet{balance: balance}
+}
+
+// Deposit increments the wallet balance by the given argument.
+func (wallet *BtcWallet) Deposit(data Bitcoin) error {
```

Comment:
Let's call the argument something like `b` or `amount`

Reply:
Okay, I'll fix it now